### ISSUE TITLE
Schema gets confused with the same name referring to same type on different types

### ISSUE THREAD

**Issue by avedetvedea** [edgedb/edgedb#4470]
I have a type **Athletes**. Both **Team** type and **Session** type have links to it, called _athletes_. If I won't rename one of them to say _members_, following error occurs: 
```
SchemaError: it is illegal to create a type union that causes a computed link 'athletes' to mix with other versions of the same link 'athletes'
```

**Comment by elprans** [comment]
I think this was fixed by #4443.  Can you try your schema on a nightly to see if that fixed the issue for you please?

**Comment by avedetvedea** [comment]
Sorry for missing that, it is working now. I never relate my problems to what is presented in your PR titles :)
